db/migrations/20220512132042_create_repository_configurations_table.up.sql

```diff
@@ -1,18 +1,97 @@
 BEGIN;
 
+--
+-- repositories
+--
+CREATE TABLE IF NOT EXISTS repositories (
+    uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE,
+    updated_at TIMESTAMP WITH TIME ZONE,
+    url VARCHAR(255) NOT NULL,
+    last_read_time TIMESTAMP WITH TIME ZONE DEFAULT NULL,
+    last_read_error VARCHAR(255) DEFAULT NULL
+);
+
+ALTER TABLE repositories
+ADD CONSTRAINT repositories_unique_url UNIQUE (url);
+
+-- ALTER TABLE ONLY repositories
+-- ADD CONSTRAINT fk_repository_configurations
+-- FOREIGN KEY (repository_configurations_uuid)
+-- REFERENCES repository_configurations(uuid)
+-- ON DELETE SET NULL;
+
+--
+-- repository_configurations
+--
 CREATE TABLE IF NOT EXISTS repository_configurations(
     uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL,
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL,
+
     name VARCHAR (255) NOT NULL,
-    url VARCHAR (255) NOT NULL,
+    -- url VARCHAR (255) NOT NULL,
```

Comment:
I thought we would keep the url in both places and lookup the repository from the repo_config's url when needed, but looking through your pr, it looks like you moved the URL there.   I think i like this, but let me think about it some more :)

Reply:
ups! I understood it in a different way! sorry! anyway, happy to change it; thanks

Reply:
resolving this as i like the direct connection